core/src/test/java/org/apache/druid/java/util/metrics/NoopSysMonitorTest.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.java.util.metrics;
+
+import org.apache.druid.java.util.emitter.service.ServiceEmitter;
+import org.junit.Assert;
+import org.junit.Assume;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class NoopSysMonitorTest
+{
+  private static final String CPU_ARCH = System.getProperty("os.arch");
+
+  @Test
+  public void testDoMonitor()
+  {
+    Assume.assumeFalse("aarch64".equals(CPU_ARCH));
```

Comment:
nit: I think it should be safe for this test to run on all architectures, but I guess there's no harm in skipping this test on aarch64

```suggestion
```

Reply:
This test fails on aarch64 since SysMonitor is the parent of NoopSysMonitor. Sigar still gets loaded even though there's no need for it. Or should there be another class that these two classes inherit from to avoid this?

Reply:
Oops I didn't notice that when I skimmed the change. The NoopSysMonitor should just implement Monitor since it is doing nothing. Or you can extend AbstractMonitor and return false in the doMonitor() method.

Reply:
But `MetricsModule` is getting the instance of `SysMonitor` through Guice while loading, and `getSysMonitor()` provides for SysMonitor. In case of peon returning NoopSysMonitor here.

Reply:
ah ok. feel free to ignore my suggestion above.

There is a way to refactor this so that sigar won't initialize on the peons, but it seems like overkill.

Reply:
I thought of adding logic in `getMonitorScheduler` during loading for the same and it seemed overkill. If you don't mind could you please share if there is other way for future references.

Reply:
If we wanted to do something like that, I would have made SysMonitor an interface and make the current class an implementation of that interface.

Another way to approach this would be to add a new annotation and inject the Monitor interface that way https://github.com/google/guice/wiki/BindingAnnotations